### PR TITLE
Filter plans by user

### LINKED ISSUE #840: RVL-952 - WebUI should allow filtering of plans by user
**Description**
In Thailand it is imperative that users of the WebUI can filter plans by user. This will be implemented using a drop down/auto complete control where the user can select a username from a list. Ideally, the filter should have a way for the user to easily click a 'me' button . This will become important as Thailand adds all staff as 'users' of the WebUI. It is expected that role based access will NOT be implemented, but filtering of plans by user (and ideally logged in user )

### PR DESCRIPTION
**Pull request by p-netm**
close #840 

Adds a filter section in manage plans page, from which selecting a user will filter out plans that are not accessible the respective user.

### CODE REVIEW

**Comment by moshthepitt** [comment]
@p-netm once you filter by user, there is no indication on the page that a filter has been upload (even if you click on "filter" again).

Oo

Was this covered in the mockups?  If not please reach out to @rowo for guidance

**Comment by moshthepitt** [comment]
@p-netm also noticed that if you press the search button before typing anything, you will end up with `?title=undefined` and an empty result page with no way to recover.  Can we prevent this?

File: src/helpers/dataLoading/plans.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { toast } from 'react-toastify';
+import { ActionCreator, AnyAction } from 'redux';
+import { OPENSRP_PLANS_BY_USER_FILTER } from '../../constants';
+import { OpenSRPService } from '../../services/opensrp';
+import store from '../../store';
+import { growl } from '../utils';
+
+/** find plans that the given OpenMRS userName has access to
+ * @param {string} userName - openMRSUserName
```
Comment (moshthepitt) [inline src/helpers/dataLoading/plans.ts:9]: We should not mention `openMRS` at all anywhere in my view.

Comment (p-netm) [reply src/helpers/dataLoading/plans.ts:9]: sawa

**Comment by p-netm** [comment]
> Seems you cannot filter by user on the FI page http://localhost:3000/focus-investigation

@moshthepitt  will this be a applied globally or on a per table basis, i.e. since we have 2 tables in ActiveFIs page.

**Comment by moshthepitt** [comment]
> @moshthepitt will this be a applied globally or on a per table basis, i.e. since we have 2 tables in ActiveFIs page.

@p-netm I think it was globally, but please check the mockups

**Comment by p-netm** [comment]
@moshthepitt Please re-review

File: src/components/forms/PractitionerForm/UserIdSelect/index.tsx [hunk 2]
```diff
@@ -91,28 +105,42 @@ export const UserIdSelect: React.FC<Props> = props => {
     return allOpenMRSUsers;
   };
 
-  /** filters out openMRs User objects that have already been mapped to an existing
-   * practitioner, this is an effort towards ensuring a 1-1 mapping between an openMRS user
-   * and a practitioner entity
+  /** depending on the value of showPracitioners; it filters out openMRs User objects that have
+   * already been mapped to an existing practitioner, this is an effort towards ensuring a 1-1
+   * mapping between an openMRS user and a practitioner entity
```
Comment (moshthepitt) [inline src/components/forms/PractitionerForm/UserIdSelect/index.tsx:110]: Please let us avoid mentioning OpenMRS anywhere.

File: src/store/ducks/opensrp/planIdsByUser/index.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
+import { Dictionary } from '@onaio/utils';
+import { AnyAction, Store } from 'redux';
+import { createSelector } from 'reselect';
+import SeamlessImmutable from 'seamless-immutable';
+import { PlanDefinition } from '../../../../configs/settings';
+
+/** the reducer name */
+export const reducerName = 'planIdsByUser';
```
Comment (moshthepitt) [inline src/store/ducks/opensrp/planIdsByUser/index.ts:8]: What's preventing us from using the already existing plans reducer module?

Comment (p-netm) [reply src/store/ducks/opensrp/planIdsByUser/index.ts:8]: the plans reducer module stores plans with the `PlanRecord` type, but the plans that we get from the filter endpoint are of `PlanDefinitionList` type and are handled by the opensrp/PlanDefinition reducer.

Comment (moshthepitt) [reply src/store/ducks/opensrp/planIdsByUser/index.ts:8]: Cool thanks, does that mean that we can use this reducer https://github.com/onaio/reveal-frontend/blob/89e8ef36c9a6f2fa036d598630bb6082ad3bfc56/src/store/ducks/opensrp/PlanDefinition/index.ts

**Comment by moshthepitt** [comment]
> We should not allow submitting the empty search box to result in http://localhost:3000/focus-investigation?title=undefined or http://localhost:3000/plans/list?title=undefined

Do we have a test to prevent a regression on this?

**Comment by p-netm** [comment]
> > We should not allow submitting the empty search box to result in http://localhost:3000/focus-investigation?title=undefined or http://localhost:3000/plans/list?title=undefined
> 
> Do we have a test to prevent a regression on this?

@moshthepitt I do not think we need one:
- There is no way to set the title value `undefined` from the frontend.
- Also there is just no feasible way to simulate such an action given that the input in the searchform is auto-consumed as soon as input starts, you either have some text or don't(an empty string).

**Comment by p-netm** [comment]
I also think it would be best to remove the search feature form the admin module pages, and add them in another PR.

**Comment by moshthepitt** [comment]
> I also think it would be best to remove the search feature form the admin module pages, and add them in another PR.

@p-netm I agree.  Please go ahead and do this.

**Comment by p-netm** [comment]
close #848

**Pull request merged**